frontend/src/api/network/notebooks.ts

```diff
@@ -43,6 +43,8 @@ const assembleNotebookAffinityAndTolerations = (
       operator: 'Exists',
     });
   } else {
+    delete resources.limits?.['nvidia.com/gpu'];
+    delete resources.requests?.['nvidia.com/gpu'];
```

Comment:
Ah yeah -- probably shouldn't mutate this either -- can we return an updated set of resources or something instead?

Mutations of variables you are passed has a long history of causing issues weirdly. Imagine a bug in a scenario completely separate of this, 2 months from now. When I do xyz the data just doesn't appear anymore.... or is different than what I did... and I have no idea why. Turns out, it was some function call in a dozen that mutated the variable. Best to always be pure about input so you don't have a headache later.